File: A-SOURCE_CODE/B-CLIENT/A-WEB/Administration/src/viewmodels/post/FindPostViewModel.ts
import {TextSearch} from "../TextSearch";
import {UnixDateRange} from "../UnixDateRange";
import {Pagination} from "../Pagination";

export class FindPostViewModel{

    // Id of post
    public id: number;

    // Owner index of post.
    public ownerIndex: number;

    // Category index of post.
    public categoryIndex: number;

    // Title of post
    public title: TextSearch;

    // Body of post
    public body: TextSearch;

    // Time when post is created.
    public created: UnixDateRange;

    // Time when post was lastly modified.
    public lastModified: UnixDateRange;

    // Pagination of records.
    public pagination: Pagination;
}

File: A-SOURCE_CODE/A-SERVICE/Administration/Shared/ViewModels/Posts/FindPostViewModel.cs
using Shared.Enumerations.Order;
using Shared.Models;

namespace Shared.ViewModels.Posts
{
    public class FindPostViewModel
    {
        /// <summary>
        ///     Post index.
        /// </summary>
        public int? Id { get; set; }

        /// <summary>
        ///     Index of post owner.
        /// </summary>
        public int? OwnerIndex { get; set; }

        /// <summary>
        ///     Index of category which post belongs to.
        /// </summary>
        public int? CategoryIndex { get; set; }

        /// <summary>
        ///     Title search option.
        /// </summary>
        public TextSearch Title { get; set; }

        /// <summary>
        ///     Body of post search.
        /// </summary>
        public TextSearch Body { get; set; }

        /// <summary>
        ///     When the post was created.
        /// </summary>
        public UnixDateRange Created { get; set; }

        /// <summary>
        /// Record sort direction.
        /// </summary>
        public SortDirection Direction { get; set; }

        /// <summary>
        /// Which property should be used for sorting.
        /// </summary>
        public PostSort Sort { get; set; }

        /// <summary>
        ///     When the post was lastly modified.
        /// </summary>
        public UnixDateRange LastModified { get; set; }

        /// <summary>
        ///     Pagination information.
        /// </summary>
        public Pagination Pagination { get; set; }
    }
}